annotation/functions.scm

```diff
@@ -401,18 +401,43 @@
     )))
 )
 
+;; Find coding Gene for a given protein
+(define-public find-coding-gene
+  (lambda (protein)
+  (cog-outgoing-set (cog-execute! (BindLink
+    (VariableNode "$g")
+    (EvaluationLink
+      (PredicateNode "expresses")
+      (ListLink
+        (VariableNode "$g")
+        protein
+      )
+    )
+    (EvaluationLink
+      (PredicateNode "expresses")
+      (ListLink
+        (VariableNode "$g")
+        protein
+      )
+    )
+  )
+  )
+)))
+
 (define-public add-mol-info
   (lambda (mol path)
   (if (string-contains (cog-name path) "R-HSA")
     (ListLink
       (MemberLink mol path)
+      (find-coding-gene mol)
```

Comment:
The `mol` argument here could be a Protein or ChEBI. Do we need to find coding genes for ChEBI?